sql/parser/sql.y

```diff
@@ -3094,18 +3098,36 @@ position_list:
 // here, and convert the SQL9x style to the generic list for further
 // processing. - thomas 2000-11-28
 substr_list:
-  a_expr substr_from substr_for { unimplemented() }
+  a_expr substr_from substr_for 
+  { 
+    $$ = Exprs{$1, $2, $3}
+  }
 | a_expr substr_for substr_from { unimplemented() }
-| a_expr substr_from { unimplemented() }
+| a_expr substr_from
+  { 
+    $$ = Exprs{$1, $2}
+  }
 | a_expr substr_for { unimplemented() }
-| expr_list { unimplemented() }
-| /* EMPTY */ {}
+| expr_list 
+  { 
+    $$ = $1
+  }
+| /* EMPTY */ 
+  {
+    $$ = nil
```

Comment:
am I reading this right that `SUBSTRING()` parses? is that just so the error gets generated at dispatch?